### PR TITLE
[timeseries] Align TimeSeriesPredictor.leaderboard API with TabularPredictor

### PR DESCRIPTION
**Pull request by shchur**
*Issue #, if available:* #4152

*Description of changes:*
- Two new arguments to `TimeSeriesPredictor.leaderboard()`:
  - `extra_metrics` allows computing additional metrics on the test set
  - `extra_info` returns hyperparameters selected by hyperparameter tuning for each model

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: timeseries/src/autogluon/timeseries/predictor.py [hunk 1]
```diff
@@ -1236,6 +1238,17 @@ def leaderboard(
...
+            If True, the leaderboard will contain an additional column `hyperparameters` with the hyperparameters used
+            by each model during training. An empty dictionary `{}` means that the model was trained with default
+            hyperparameters.
+        extra_metrics : List[Union[str, TimeSeriesScorer]], optional
+            A list of metrics to calculate scores for and include in the output DataFrame.
+
+            Only valid when `data` is specified. The scores refer to the scores on `data` (same data as used to
+            calculate the `score_test` column).
+
+            This list can contain any values which would also be valid for `eval_metric` when creating a `TimeSeriesPredictor`.
```
Comment (Innixma) [inline timeseries/src/autogluon/timeseries/predictor.py:1251]: nit: Mention the column name they will have in the output. I think this is also in the Tabular docstring

File: .github/workflow_scripts/lint_check.sh [hunk 2]
```diff
@@ -22,6 +22,6 @@ function lint_check_all {
     # lint_check tabular
 }
 
-bandit -r multimodal/src -ll
+bandit -r multimodal/src -ll --skip B614
```
Comment (shchur) [inline .github/workflow_scripts/lint_check.sh:25]: @Innixma skipping the new rule that marks `torch.save` / `torch.load` as unsafe to unblock the CI
